src/openshift/components/CryostatContainer.tsx

```diff
@@ -368,6 +369,6 @@ const NamespacedContainer: React.FC<{ searchNamespace: string; children: React.R
 
 export const CryostatContainer: React.FC = ({ children }) => {
   const [namespace] = useActiveNamespace();
-
+  checkNavHighlighting();
```

Comment:
Should this be wrapped in an effect hook with no dependencies?

```typescript
React.useEffect(() => checkNavHighlighting(), []);
```

When placed directly into the component body like this, it'll be executed before the component renders and also every time it re-renders. If it's inside of an effect hook then the first execution will only happen after the component renders for the first time. Since the hook has no dependencies it will only re-render when the whole component is unmounted and remounted.

Reply:
https://stackoverflow.com/questions/58579426/in-useeffect-whats-the-difference-between-providing-no-dependency-array-and-an

Reply:
Good call, yes I'll do that instead